### PR TITLE
Always close input streams, output streams etc

### PR DESCRIPTION
**Pull request by jonescc**
Fixes https://github.com/aodn/geoserver-build/issues/137

### CODE REVIEW

File: src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/FeatureSourceIndexReader.java [hunk 1]
```diff
@@ -62,25 +60,15 @@ public URIList getUriList(String profile, String timeField, String urlField, GoG
...
 
-        SimpleFeatureIterator iterator = null;
-        try {
-            iterator = getFeatures(typeName, query);
-        } catch (IOException e) {
-            throw new GoGoDuckException(e.getMessage());
-        }
-
-        try {
+        try (SimpleFeatureIterator iterator = getFeatures(typeName, query)) {
```
Comment (julian1) [inline src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/FeatureSourceIndexReader.java]: Is it not possible to rewrite this using the 

```
for( E element : list ) { 
```
idiom in java?

Comment (julian1) [inline src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/FeatureSourceIndexReader.java]: Eg. http://stackoverflow.com/questions/18410035/ways-to-iterate-over-a-list-in-java

File: src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/UserLog.java [hunk 2]
```diff
@@ -1,51 +0,0 @@
...
-        this.userLogFileWriter = null;
-    }
-
-    UserLog(String logFile) {
-        try {
-            this.userLogFileWriter = new FileWriter(logFile);
-        }
-        catch (IOException e) {
-            this.userLogFileWriter = null;
-            logger.info(String.format("Exception while opening user log at '%s', continuing anyway", logFile));
```
Comment (julian1) [inline src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/UserLog.java]: fun!

File: src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/FeatureSourceIndexReader.java (hunk 1 above)
Comment (jonescc) [inline src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/FeatureSourceIndexReader.java]: can't use a for with SimplefeatureCollection - its a geotools collection not a real java collection.

**Pull request merged**